src/mock/index.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * @param {string} status: - passing in "resolve" will ensure the operation is resolved all
+ *                           the way through.
+ *                         - passing in "reject" will ensure the operation fails.
+ * @param {number?} timeout: time in milliseconds, mock fetching request time.
+ * @returns {void}
+ * */
+import * as data from "./data.json";
+
+export const mockFetch = (
+  status: "resolve" | "reject",
+  timeout = 0
+): Promise<any> => {
+  return new Promise((resolve, _) => {
+    setTimeout(() => {
+      setTimeout(() => {
+        resolve({
+          //
+          json: () =>
+            new Promise((resolve, reject) => {
+              setTimeout(() => {
+                if (status === "reject") {
```

Comment:
Standard practice is to handle the success case first, then deal with errors after